beginner_source/hybrid_frontend/introduction_to_hybrid_frontend_tutorial.py

```diff
@@ -0,0 +1,526 @@
+# -*- coding: utf-8 -*-
+"""
+Introduction to the Hybrid Frontend
+===================================
+**Author:** `Matthew Inkawhich <https://github.com/MatthewInkawhich>`_
+
+In this tutorial, we will experiment with some of the key features of
+the PyTorch hybrid frontend.
+
+This flexible programming model enables the co-existance and seamless
+interaction between an imperative eager interface designed for
+experimentation and easy debugging, and a declarative graph mode in
+which models can be optimized for performance and exported to
+large-scale production environments.
+
+"""
+
+######################################################################
+# Eager Mode
+# ----------
+#
+# PyTorch's eager mode is the familiar PyTorch interface. Eager mode is
+# easy because it is simply idiomatic Python. Because of the imperative
+# nature of the mode, we can use our favorite Python debugging tools such
+# as ``pdb`` and ``print`` statements. This mode is designed for rapid
+# model prototyping and ambitious research efforts.
+#
+# However, it is difficult to optimize models represented in eager mode
+# for large-scale production and deployment.
+#
+# Graph Mode
+# ----------
+#
+# The graph mode creates a deferred model representation, which is a
+# common methodology in many deep learning frameworks. In graph mode, you
+# write a model definition in Python, which is run later on in the Caffe2
+# C++ runtime environment. Having a deferred representation has numerous
+# benefits, including:
+#
+#   - *Model simplicity*: Model representation is simpler and more restricted
+#   - *Optimization*: We can apply various optimization techniques to create
+#     more efficient deployable models
+#   - *Improved performance*: We can leverage parallel and out-of-order
+#     execution, and target highly optimized hardware architectures
+#   - *Easy export*: Exporting a serialized model is a breeze with ONNX
+#
+# .. figure:: /_static/img/hybrid_frontend/pytorch_workflow_small.jpg
+#    :alt: workflow
+
+
+
+
+######################################################################
+# JIT Compiler
+# ------------
+#
+# To accomodate to as many workflows as possible, PyTorch not only
+# provides both an eager mode and a graph mode, but a transition mechanism
+# between them: ``torch.jit``. This enables the use of the newest
```

Comment:
All of the script mode is contained under ``torch.jit`` so calling it the transition mechanism is not very accurate. We can say,  

""PyTorch provides mechanisms for incrementally converting eager-mode code into Torch Script."